salt/grains/core.py

```diff
@@ -1678,10 +1678,11 @@ def fqdns():
     addresses = salt.utils.network.ip_addrs(include_loopback=False, interface_data=_INTERFACES)
     addresses.extend(salt.utils.network.ip_addrs6(include_loopback=False, interface_data=_INTERFACES))
 
-    try:
-        fqdns = map(lambda ip: socket.gethostbyaddr(ip)[0], addresses)
-    except (socket.error, socket.herror, socket.gaierror, socket.timeout) as e:
-        log.error("Exception during resolving address: " + e)
+    for ip in addresses:
+        try:
+            fqdns.add(socket.gethostbyaddr(ip)[0])
+        except (socket.error, socket.herror, socket.gaierror, socket.timeout) as e:
+            log.error("Exception during resolving address: " + str(e))
     
     grains['fqdns'] = list(set(fqdns))
```

Comment:
Last nitpick: `fqdns` is already instantiated as a set() so no need to reconvert it to set() again.

Reply:
Right: fixed :)